### ISSUE TITLE
Question: initial map and loss problem in yolov5 and yolov8

### ISSUE THREAD

**Issue by nakuda1** [open-mmlab/mmyolo#656]
### Prerequisite

- [X] I have searched [the existing and past issues](https://github.com/open-mmlab/mmyolo/issues) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmyolo.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version](https://github.com/open-mmlab/mmyolo).

### 🐞 Describe the bug

Sorry to bother you. I have two problems below.

1. I set the same parameters in yolov5 and yolov8 to train the tutorial cat dataset(aroud 200 pictures), why the intial maps are so different. it seems like yolov5 does not load the pth. In yolov5, my model inherits the yolov5_s-v61_syncbn_8xb16-300e_coco.py. in yolov8, my model inherits the yolov8_s_syncbn_fast_8xb16-500e_coco.py.

![7a995161a8d298a457a2a09d30380e7](https://user-images.githubusercontent.com/122974900/224637733-167baaac-33ff-4ddb-acb2-66ab2766b3e4.png)

2.  And when I use yolov8, why the loss is so high. When I see the yolov8 of mmyolo log, the loss is also so high. Maybe it is not convenient for experiment comparison. 

![image](https://user-images.githubusercontent.com/122974900/224638418-06cce0fc-0719-451d-89d5-6d3a3aee3b33.png)

---------------------------------------------------------------------------------------------------------------------------------------

**### yolov5 change backbone code**:
_base_ = './yolov5_s-v61_syncbn_8xb16-300e_coco.py'
#之前是继承yolov5/yolov5_s-v61_1xb2-1e_coco128.py,现在之前继承第一个父类，不加深度了

custom_imports = dict(imports=['mmcls.models'], allow_failed_imports=False)

widen_factor = 1.0
channels = [40, 80, 160]

model = dict(
    backbone=dict(
        _delete_=True,  # 将 _base_ 中关于 backbone 的字段删除
        type='mmcls.TIMMBackbone',  # 使用 mmcls 中的 timm 主干网络
        model_name='ghostnet_100',  # 使用 TIMM 中的 mobilevitv2_050
        features_only=True,
        pretrained=True,
        out_indices=(2, 3, 4)),
    neck=dict(
        type='YOLOv5PAFPN',
        widen_factor=widen_factor,
        in_channels=channels,  # 注意：mobilevitv2_050 输出的3个通道是 [128, 192, 256]，和原先的 yolov5-s neck 不匹配，需要更改
        out_channels=channels),
    bbox_head=dict(
        type='YOLOv5Head',
        head_module=dict(
            type='YOLOv5HeadModule',
            in_channels=channels,  # head 部分输入通道也要做相应更改
            widen_factor=widen_factor))
)

#继承的代码可以直接加在这个下面就可以方便以后修改了

config_ghostnet_100 = f"""
_base_=\'{_base_}\'
widen_factor={widen_factor}
model={model}
#加入可视化,目前先关闭，等模型最后构建好再加入可视化, cmd命令要启动wandb
visualizer = dict(vis_backends=[dict(type='LocalVisBackend'), dict(type='WandbVisBackend')])

max_epochs= 400  # 训练的最大 epoch
data_root = './data/cat/'  # 数据集目录的绝对路径

anchors = [
    [(68, 69), (154, 91), (143, 162)],  # P3/8
    [(242, 160), (189, 287), (391, 207)],  # P4/16
    [(353, 337), (539, 341), (443, 432)]  # P5/32
]

#work_dir = './work_dirs/yolov5_s_ghostnet_100-v61_1xb2-1e_coco128.py'

train_batch_size_per_gpu = 32
train_num_workers = 4  # 推荐使用 train_num_workers = nGPU x 4

save_epoch_intervals = 2  # 每 interval 轮迭代进行一次保存一次权重

base_lr = _base_.base_lr / 8

class_name = ('cat', )  # 根据 class_with_id.txt 类别信息，设置 class_name
num_classes = len(class_name)
metainfo = dict(
    classes=class_name, # 注意： 在新版本中已经换成了小写
    palette=[(220, 20, 60)]  # 画图时候的颜色，随便设置即可
)

train_cfg = dict(
    max_epochs=max_epochs,
    val_begin=20,  # 第几个 epoch 后验证，这里设置 20 是因为前 20 个 epoch 精度不高，测试意义不大，故跳过
    val_interval=save_epoch_intervals,  # 每 val_interval 轮迭代进行一次测试评估
    dynamic_intervals=[(max_epochs - _base_.num_last_epochs, 1)]
)

train_dataloader = dict(
    batch_size=train_batch_size_per_gpu,
    num_workers=train_num_workers,
    dataset=dict(
        _delete_=True,
        type='RepeatDataset',
        # 数据量太少的话，可以使用 RepeatDataset ，在每个 epoch 内重复当前数据集 n 次，这里设置 5 是重复 5 次
        times=5,
        dataset=dict(
            type=_base_.dataset_type,
            data_root=data_root,
            metainfo=metainfo,
            ann_file='annotations/trainval.json',
            data_prefix=dict(img='images/'),
            filter_cfg=dict(filter_empty_gt=False, min_size=32),
            pipeline=_base_.train_pipeline)))

val_dataloader = dict(
    dataset=dict(
        metainfo=metainfo,
        data_root=data_root,
        ann_file='annotations/trainval.json',
        data_prefix=dict(img='images/')))

test_dataloader = val_dataloader

val_evaluator = dict(ann_file=data_root + 'annotations/trainval.json')
test_evaluator = val_evaluator

optim_wrapper = dict(optimizer=dict(lr=base_lr))

default_hooks = dict(
    # 设置间隔多少个 epoch 保存模型，以及保存模型最多几个，`save_best` 是另外保存最佳模型（推荐）
    checkpoint=dict(
        type='CheckpointHook',
        interval=save_epoch_intervals,
        max_keep_ckpts=5,
        save_best='auto'),
    param_scheduler=dict(max_epochs=max_epochs),
    # logger 输出的间隔
    logger=dict(type='LoggerHook', interval=10))

custom_hooks = [
    dict(
        type='EMAHook',
        ema_type='ExpMomentumEMA',
        momentum=0.0001,
        update_buffers=True,
        strict_load=False,
        priority=49),
    dict(
        type='mmdet.PipelineSwitchHook',
        switch_epoch=max_epochs - _base_.num_last_epochs,
        switch_pipeline=_base_.train_pipeline_stage2)
]

"""

with open('./configs/yolov5/yolov5_s_ghostnet_100-v61_1xb2-1e_coco128.py', 'w') as f:
    f.write(config_ghostnet_100)

---------------------------------------------------------------------------------------------------------------------------
### **yolov8 change model code：**

_base_ = './yolov8_s_syncbn_fast_8xb16-500e_coco.py'
#之前是继承yolov8/yolov8_s-v61_1xb2-1e_coco128.py,现在之前继承第一个父类，不加深度了

custom_imports = dict(imports=['mmcls.models'], allow_failed_imports=False)

widen_factor = 1.0
channels = [40, 80, 160]

model = dict(
    backbone=dict(
        _delete_=True,  # 将 _base_ 中关于 backbone 的字段删除
        type='mmcls.TIMMBackbone',  # 使用 mmcls 中的 timm 主干网络
        model_name='ghostnet_100',  # 使用 TIMM 中的 mobilevitv2_050
        features_only=True,
        pretrained=True,
        out_indices=(2, 3, 4)),
    neck=dict(
        type='YOLOv8PAFPN',
        widen_factor=widen_factor,
        in_channels=channels,  # 注意：mobilevitv2_050 输出的3个通道是 [128, 192, 256]，和原先的 yolov5-s neck 不匹配，需要更改
        out_channels=channels),
    bbox_head=dict(
        type='YOLOv8Head',
        head_module=dict(
            type='YOLOv8HeadModule',
            in_channels=channels,  # head 部分输入通道也要做相应更改
            widen_factor=widen_factor))
)

#继承的代码可以直接加在这个下面就可以方便以后修改了

config_ghostnet_100 = f"""
_base_=\'{_base_}\'
widen_factor={widen_factor}
model={model}
#加入可视化,目前先关闭，等模型最后构建好再加入可视化, cmd命令要启动wandb
visualizer = dict(vis_backends=[dict(type='LocalVisBackend'), dict(type='WandbVisBackend')])

max_epochs= 350  # 训练的最大 epoch
data_root = './data/cat/'  # 数据集目录的绝对路径

#work_dir = './work_dirs/yolov5_s_ghostnet_100-v61_1xb2-1e_coco128.py'

train_batch_size_per_gpu = 32
train_num_workers = 4  # 推荐使用 train_num_workers = nGPU x 4

save_epoch_intervals = 2  # 每 interval 轮迭代进行一次保存一次权重

base_lr = _base_.base_lr / 8

class_name = ('cat', )  # 根据 class_with_id.txt 类别信息，设置 class_name
num_classes = len(class_name)
metainfo = dict(
    classes=class_name, # 注意： 在新版本中已经换成了小写
    palette=[(220, 20, 60)]  # 画图时候的颜色，随便设置即可
)

train_cfg = dict(
    max_epochs=max_epochs,
    val_begin=20,  # 第几个 epoch 后验证，这里设置 20 是因为前 20 个 epoch 精度不高，测试意义不大，故跳过
    val_interval=save_epoch_intervals,  # 每 val_interval 轮迭代进行一次测试评估
    dynamic_intervals=[(max_epochs - _base_.num_last_epochs, 1)]
)

train_dataloader = dict(
    batch_size=train_batch_size_per_gpu,
    num_workers=train_num_workers,
    dataset=dict(
        _delete_=True,
        type='RepeatDataset',
        # 数据量太少的话，可以使用 RepeatDataset ，在每个 epoch 内重复当前数据集 n 次，这里设置 5 是重复 5 次
        times=5,
        dataset=dict(
            type=_base_.dataset_type,
            data_root=data_root,
            metainfo=metainfo,
            ann_file='annotations/trainval.json',
            data_prefix=dict(img='images/'),
            filter_cfg=dict(filter_empty_gt=False, min_size=32),
            pipeline=_base_.train_pipeline)))

val_dataloader = dict(
    dataset=dict(
        metainfo=metainfo,
        data_root=data_root,
        ann_file='annotations/trainval.json',
        data_prefix=dict(img='images/')))

test_dataloader = val_dataloader

val_evaluator = dict(ann_file=data_root + 'annotations/trainval.json')
test_evaluator = val_evaluator

optim_wrapper = dict(optimizer=dict(lr=base_lr))

default_hooks = dict(
    # 设置间隔多少个 epoch 保存模型，以及保存模型最多几个，`save_best` 是另外保存最佳模型（推荐）
    checkpoint=dict(
        type='CheckpointHook',
        interval=save_epoch_intervals,
        max_keep_ckpts=5,
        save_best='auto'),
    param_scheduler=dict(max_epochs=max_epochs),
    # logger 输出的间隔
    logger=dict(type='LoggerHook', interval=10))

custom_hooks = [
    dict(
        type='EMAHook',
        ema_type='ExpMomentumEMA',
        momentum=0.0001,
        update_buffers=True,
        strict_load=False,
        priority=49),
    dict(
        type='mmdet.PipelineSwitchHook',
        switch_epoch=max_epochs - _base_.num_last_epochs,
        switch_pipeline=_base_.train_pipeline_stage2)
]

with open('./configs/yolov8/yolov8_s_ghostnet_100-v61_1xb2-1e_coco128.py', 'w') as f:
    f.write(config_ghostnet_100)

### Environment

the mmyolo tutorial environment.

**Comment by hhaAndroid** [comment]
@nakuda1 If your configuration is inherited from `yolov5_s-v61_syncbn_8xb16-300e_coco.py`, indeed no pre-training weights are loaded. You can refer to the cat configuration if you want to load it.
Different algorithm loss calculation is not the same, so the value of the loss is not comparable, I suggest you directly compare mAP
